Refactor: move into package + add caching of files

RDIL opened the pull request:
I used the namespace `me.param` since that is your domain
local module names should be uppercase
### also this may break stuff

RDIL committed:
Refactor: move into package

RDIL committed:
Rename algorithm.py to me/param/Algorithm.py

RDIL committed:
Rename constants.py to me/param/Constants.py

RDIL committed:
Update App.py

RDIL commented:
@paramt may want to remove the requirement for test job to pass

RDIL committed:
Update App.py

RDIL committed:
Create Cache.py

RDIL committed:
adapt to use cache

RDIL committed:
Adapt methods

RDIL committed:
Update App.py

RDIL committed:
Update Cache.py

RDIL commented:
Okay done

RDIL committed:
unused

paramt commented:
I apologize for my ignorance, but I'm having a hard time understanding the file caching logic and how it helps... [Also, as far as I'm aware, all python imports are lower-case](https://www.python.org/dev/peps/pep-0008/#package-and-module-names).

RDIL commented:
Cache allows for the code to resolve the file it is reading/writing from faster, to speed up the process

RDIL committed:
Rename Algorithm.py to algorithm.py

RDIL committed:
Update and rename Cache.py to cache.py

RDIL committed:
Rename Constants.py to constants.py

RDIL committed:
Update and rename App.py to app.py

paramt committed:
Move app.py into /src

paramt committed:
Move cache.py into /src

paramt committed:
Move algorithm.py into /src

paramt committed:
Move constants.py into /src

paramt commented on src/app.py:
```diff
@@ -4,14 +4,21 @@
 
-reddit = praw.Reddit('MemeAdviser')
+# check if this is being imported or not
+if __name__ not "__main__":
```
```suggestion
if __name__ != "__main__":
```
`SyntaxError: invalid syntax`

paramt commented on src/app.py:
```diff
@@ -4,14 +4,21 @@
 
-with open("../subscribed.txt", "r") as f:
-    subscribed = list(filter(None, f.read().split("\n")))
+replied_file = Cache.AbstractFile("../replied.txt")
```
```suggestion
replied_file = cache.AbstractFile("../replied.txt")
```

paramt commented on src/app.py:
```diff
@@ -4,14 +4,21 @@
-with open("../subscribed.txt", "r") as f:
-    subscribed = list(filter(None, f.read().split("\n")))
+replied_file = Cache.AbstractFile("../replied.txt")
+subscribed_file = Cache.AbstractFile("../subscribed.txt")
```
```suggestion
subscribed_file = cache.AbstractFile("../subscribed.txt")
```

paramt commented on src/app.py:
```diff
@@ -4,14 +4,21 @@
-    subscribed = list(filter(None, f.read().split("\n")))
+replied_file = Cache.AbstractFile("../replied.txt")
+subscribed_file = Cache.AbstractFile("../subscribed.txt")
+fileLoadCache = Cache.FileCache()
```
```suggestion
fileLoadCache = cache.FileCache()
```

paramt commented on src/cache.py:
```diff
@@ -0,0 +1,55 @@
+# |==========================================|
+
+# Taken from a project that may become open source soon, please keep the copyright header
+
```
```suggestion

from io import TextIOWrapper

```

`NameError: name 'TextIOWrapper' is not defined`

RDIL replied:
Huh, should be in memory by default

paramt commented on src/cache.py:
```diff
@@ -0,0 +1,55 @@
+        finally:
+            return True
+
+    def clear_file(file: str) -> None:
```
```suggestion
    def clear_file(self, file: str) -> None:
```
`TypeError: clear_file() takes 1 positional argument but 2 were given`

paramt commented on src/app.py:
```diff
@@ -34,9 +41,8 @@
-            for post_id in replied:
-                f.write(post_id + "\n")
+        for post_id in replied:
+            replied.write_to_file(post_id + "\n")
```
```
Traceback (most recent call last):
  File "app.py", line 45, in <module>
    replied.write_to_file(post_id + "\n")
AttributeError: 'list' object has no attribute 'write_to_file'
```

RDIL replied:
Ah, I can fix that.

paramt reviewed (changes requested):
The script isn't updating `replied.txt`, which needs to be fixed before merging. I've highlighted several minor syntax errors as well, but other than that it looks good. After you apply the changes, I'll have to test it again before merging the PR

paramt committed:
Update src/app.py

Co-Authored-By: RDIL <me@rdil.rocks>

paramt committed:
Update src/app.py

Co-Authored-By: RDIL <me@rdil.rocks>

paramt committed:
Update src/app.py

Co-Authored-By: RDIL <me@rdil.rocks>

paramt committed:
Update src/app.py

Co-Authored-By: RDIL <me@rdil.rocks>

paramt committed:
Update src/cache.py

Co-Authored-By: RDIL <me@rdil.rocks>

paramt committed:
Update src/cache.py

Co-Authored-By: RDIL <me@rdil.rocks>

RDIL committed:
Work to fix error

RDIL committed:
fixed issues (I hope) + add pydocs

Signed-off-by: Reece <me@rdil.rocks>

paramt commented on src/cache.py:
```diff
@@ -0,0 +1,121 @@
+        :param file: the @AbstractFile to refresh
+        :return: some list thing
+        """
+        with open(file.get_name(), mode="r") as filehandler:
```
```
Traceback (most recent call last):
  File "app.py", line 18, in <module>
    replied = fileLoadCache.refresh(replied_file.get_name())
  File "cache.py", line 51, in refresh
    with open(file.get_name(), mode="r") as filehandler:
AttributeError: 'str' object has no attribute 'get_name'
```

RDIL replied:
oh right smh

RDIL replied:
which python version?

paramt replied:
2.7, but I can upgrade if it's required

RDIL replied:
sir
your going to need latest

paramt replied:
I can run 3.6.7

RDIL replied:
3.7.2 is prob best

paramt replied:
Hmm.. would 3.7.1 work?

paramt replied:
@RDIL 3.7.2 is now installed on the server

RDIL replied:
Yeah that should work - thanks!

paramt committed:
Refactor code

- Create .gitignore
- Move replied.txt and subscribed.txt inside /data
- Add test.py
- Move app.py code into main function

paramt committed:
Merge master branch

paramt committed:
Update README to reflect changes

RDIL committed:
Merge branch 'master' into RDIL-patch-1

RDIL committed:
Finished fixing the code I hope

paramt committed:
Fix README code block 

The apostrophe in [YOUR APP'S ID] was formatted as a single quote and highlighted as a string

RDIL commented:
Anything else not working?

paramt commented:
@RDIL Unfortunately, I won't be able to test the code until Sunday evening

RDIL commented:
No worries

paramt commented on src/app.py:
```diff
@@ -0,0 +1,98 @@
+            # Update replied.txt
+            replied.append(submission.id)
+            for post_id in replied:
+                replied.write_to_file(post_id + "\n")
```
```
Traceback (most recent call last):
  File "app.py", line 97, in <module>
    main(constants.Thresholds)
  File "app.py", line 42, in main
    replied.write_to_file(post_id + "\n")
AttributeError: 'list' object has no attribute 'write_to_file'
```

RDIL replied:
Something is happening with a variable conflict

paramt replied:
I can't seem to find any conflicts... `replied` is only declared once, on line 15
It seems that `write_to_file` is a method of the `FileCache` class, but 
```python
replied = fileLoadCache.refresh(replied_file)
``` 

returns a list

RDIL replied:
Ah, that’s it.  Any idea how to work around this?

paramt replied:
I don't really understand `cache.py` but we could replace '`replied.write_to_file()` with 

```python 
with open("/data/replied.txt", "w") as f: 
    for post_id in replied: 
        f.write(post_id + "\n")
```

like it was before... Or does that just make the whole thing pointless?

RDIL replied:
I need to take a break from this PR for my mental health, I’ll reopen it once I have a fix.

paramt replied:
Take your time, don't stress about it

paramt committed:
Change 'Contact me' link

Link to reddit private message instead of email
